core/templates/dev/head/components/skills-mastery-list/skills-mastery-list.directive.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the skills mastery list.
+ */
+
+require('components/concept-card/concept-card.directive.ts');
+require('components/skills-mastery-list/skills-mastery-list.constants.ts');
+require('domain/utilities/UrlInterpolationService.ts');
+require('services/UserService.ts');
+
+angular.module('oppia').directive('skillsMasteryList', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      bindToController: {
+        getDegreesOfMastery: '&degreesOfMastery',
+        getSkillDescriptions: '&skillDescriptions'
+      },
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/components/skills-mastery-list/skills-mastery-list.directive.html'),
+      controllerAs: '$ctrl',
+      controller: [
+        '$scope', '$uibModal', 'UserService',
+        'MASTERY_CUTOFF', 'MASTERY_COLORS',
+        function(
+            $scope, $uibModal, UserService,
+            MASTERY_CUTOFF, MASTERY_COLORS) {
+          var ctrl = this;
+          ctrl.userIsLoggedIn = null;
+          UserService.getUserInfoAsync().then(function(userInfo) {
+            ctrl.canCreateCollections = userInfo.canCreateCollections();
```

Comment:
make it userCanCreateCollection*

Reply:
Uh, actually this is not needed for this PR. I have removed that.